src/metpy/calc/kinematics.py

```diff
@@ -567,7 +567,18 @@ def frontogenesis(potential_temperature, u, v, dx=None, dy=None, x_dim=-1, y_dim
 
     # Compute the angle (beta) between the wind field and the gradient of potential temperature
     psi = 0.5 * np.arctan2(shrd, strd)
-    beta = np.arcsin((-ddx_theta * np.cos(psi) - ddy_theta * np.sin(psi)) / mag_theta)
+    arg = (-ddx_theta * np.cos(psi) - ddy_theta * np.sin(psi)) / mag_theta
+
+    # A few problems may occur when calculating the argument to the arcsin function.
+    # First, we may have divided by zero, since a constant theta field would mean
+    # mag_theta is zero.  To counter this, we set the argument to zero in this case.
+    # Second, due to round-off error, the argument may be slightly outside the domain
+    # of arcsin.  To counter this, we use np.clip to force the argument to be an
+    # acceptable value.  With these adjustments, we can make sure beta doesn't end up
+    # with nans somewhere.
+    arg[mag_theta == 0] = 0
+    arg = np.clip(arg, -1, 1)
+    beta = np.arcsin(arg)
 
     return 0.5 * mag_theta * (tdef * np.cos(2 * beta) - div)
```

Comment:
If I remember the cosine double-angle identities correctly, this should be equivalent:
```suggestion

    # cos(2*theta) = cos(theta)**2 - sin(theta)**2 = 1 - 2 * sin(theta)**2
    # second equality uses cos(theta)**2 + sin(theta)**2 = 1
    # sin(arcsin(arg)) == arg for all -1 <= arg <= 1
    return 0.5 * mag_theta * (tdef * (1 - 2 * arg**2) - div)
```
The `np.clip` provides an opportunity to correct roundoff error, but we might be able to skip it and still get decent answers.  (Fortunately the errors are near one rather than zero, so going a bit past doesn't introduce sign errors, so the relevant thing is whether this roundoff error is smaller than the discretization error in `mag_theta` and `div`).

If you want to declare the trigonometric identities out-of-scope for this PR, that is entirely fine.

Reply:
I just ran the same algebra and came to the same conclusion. I very much prefer this formulation. Using `clip` seems like a crutch for poor numerics and a measure of last resort, rather than doing algebra that reduces to something mathematically identical, but without the domain issues.

Reply:
Ok, thanks to [odyssey and herbie](https://github.com/herbie-fp/odyssey?tab=readme-ov-file#running-odyssey) it looks like my algebra introduces some more error (likely due to catastrophic cancellation in `tdef - div`).  But I would like to see the replacement of the cos(arcsin()) formulation with the identity suggested by @Dwesl (and eliminating the need for `clip`).

Reply:
Sounds good!